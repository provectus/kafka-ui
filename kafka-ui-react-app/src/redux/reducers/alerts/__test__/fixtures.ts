export const failurePayload1 = {
  title: 'title',
  message: 'message',
  subject: 'topic-1',
};

export const failurePayload2 = {
  ...failurePayload1,
  subject: 'topic-2',
};
