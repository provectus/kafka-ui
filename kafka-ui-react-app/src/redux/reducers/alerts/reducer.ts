import { getType } from 'typesafe-actions';
import { dismissAlert } from 'redux/actions';
import { Action, AlertsState } from 'redux/interfaces';

import { addError, removeAlert } from './utils';

export const initialState: AlertsState = {};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): AlertsState => {
  const { type } = action;

  if (type.endsWith('__FAILURE')) return addError(state, action);

  if (type === getType(dismissAlert)) {
    return removeAlert(state, action);
  }

  return state;
};

export default reducer;
