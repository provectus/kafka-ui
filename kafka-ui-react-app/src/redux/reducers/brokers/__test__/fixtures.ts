export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com' },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com' },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  zooKeeperStatus: 1,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

export const initialBrokersReducerState = {
  items: brokersPayload,
  brokerCount: 2,
  zooKeeperStatus: 1,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 1111, segmentCount: 333 },
    { brokerId: 1, segmentSize: 2222, segmentCount: 444 },
  ],
  version: '2.2.1',
};

export const updatedBrokersReducerState = {
  items: brokersPayload,
  brokerCount: 2,
  zooKeeperStatus: 1,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};
