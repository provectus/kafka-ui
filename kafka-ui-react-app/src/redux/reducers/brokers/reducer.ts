import { Action, BrokersState, ZooKeeperStatus } from 'redux/interfaces';
import { ClusterStats } from 'generated-sources';

export const initialState: BrokersState = {
  items: [],
  brokerCount: 0,
  zooKeeperStatus: ZooKeeperStatus.offline,
  activeControllers: 0,
  onlinePartitionCount: 0,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 0,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [],
};

const updateBrokerSegmentSize = (
  state: BrokersState,
  payload: ClusterStats
) => {
  const brokers = state.items;
  const { diskUsage } = payload;

  const items = brokers.map((broker) => {
    const brokerMetrics =
      diskUsage && diskUsage.find(({ brokerId }) => brokerId === broker.id);
    if (brokerMetrics !== undefined) {
      return { ...broker, ...brokerMetrics };
    }
    return broker;
  });

  return { ...state, items, ...payload };
};

const reducer = (state = initialState, action: Action): BrokersState => {
  switch (action.type) {
    case 'GET_BROKERS__REQUEST':
      return initialState;
    case 'GET_BROKERS__SUCCESS':
      return {
        ...state,
        items: action.payload,
      };
    case 'GET_CLUSTER_STATUS__SUCCESS':
      return updateBrokerSegmentSize(state, action.payload);
    default:
      return state;
  }
};

export default reducer;
