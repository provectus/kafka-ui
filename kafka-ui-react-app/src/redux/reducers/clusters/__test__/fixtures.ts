import { Cluster, ServerStatus } from 'generated-sources';

export const onlineClusterPayload: Cluster = {
  name: 'secondLocal',
  defaultCluster: true,
  status: ServerStatus.Online,
  brokerCount: 1,
  onlinePartitionCount: 6,
  topicCount: 3,
  bytesInPerSec: 1.55,
  bytesOutPerSec: 9.314,
};
export const offlineClusterPayload: Cluster = {
  name: 'local',
  defaultCluster: false,
  status: ServerStatus.Offline,
  brokerCount: 1,
  onlinePartitionCount: 2,
  topicCount: 2,
  bytesInPerSec: 3.42,
  bytesOutPerSec: 4.14,
};

export const clustersPayload: Cluster[] = [
  onlineClusterPayload,
  offlineClusterPayload,
];
