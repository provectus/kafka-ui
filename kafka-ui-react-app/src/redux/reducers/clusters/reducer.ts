import { Action } from 'redux/interfaces';
import { Cluster } from 'generated-sources';

export const initialState: Cluster[] = [];

const reducer = (state = initialState, action: Action): Cluster[] => {
  switch (action.type) {
    case 'GET_CLUSTERS__SUCCESS':
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
