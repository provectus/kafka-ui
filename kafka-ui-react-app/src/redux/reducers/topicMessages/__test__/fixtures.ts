import {
  TopicMessage,
  TopicMessageConsuming,
  TopicMessageTimestampTypeEnum,
} from 'generated-sources';

export const topicMessagePayload: TopicMessage = {
  partition: 29,
  offset: 14,
  timestamp: new Date('2021-07-21T23:25:14.865Z'),
  timestampType: TopicMessageTimestampTypeEnum.CREATE_TIME,
  key: 'schema-registry',
  headers: {},
  content:
    '{"host":"schemaregistry1","port":8085,"master_eligibility":true,"scheme":"http","version":1}',
};

export const topicMessagePayloadV2: TopicMessage = {
  ...topicMessagePayload,
  partition: 28,
  offset: 88,
};

export const topicMessagesMetaPayload: TopicMessageConsuming = {
  bytesConsumed: 1830,
  elapsedMs: 440,
  messagesConsumed: 2301,
  isCancelled: false,
};
