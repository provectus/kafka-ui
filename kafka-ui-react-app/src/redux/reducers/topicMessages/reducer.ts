import { Action, TopicMessagesState } from 'redux/interfaces';
import { getType } from 'typesafe-actions';
import * as actions from 'redux/actions';

export const initialState: TopicMessagesState = {
  messages: [],
  meta: {
    bytesConsumed: 0,
    elapsedMs: 0,
    messagesConsumed: 0,
    isCancelled: false,
  },
  isFetching: false,
};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): TopicMessagesState => {
  switch (action.type) {
    case getType(actions.addTopicMessage): {
      return {
        ...state,
        messages: [...state.messages, action.payload],
      };
    }
    case getType(actions.resetTopicMessages):
      return initialState;
    case getType(actions.updateTopicMessagesPhase):
      return {
        ...state,
        phase: action.payload,
      };
    case getType(actions.updateTopicMessagesMeta):
      return {
        ...state,
        meta: action.payload,
      };
    case getType(actions.setTopicMessagesFetchingStatus):
      return {
        ...state,
        isFetching: action.payload,
      };
    case getType(actions.clearMessagesTopicAction.success):
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
};

export default reducer;
