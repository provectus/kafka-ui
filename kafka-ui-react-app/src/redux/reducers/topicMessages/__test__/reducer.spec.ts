import {
  addTopicMessage,
  fetchSchemaVersionsAction,
  resetTopicMessages,
  updateTopicMessagesMeta,
  updateTopicMessagesPhase,
} from 'redux/actions';
import reducer, { initialState } from 'redux/reducers/topicMessages/reducer';

import { topicMessagePayload, topicMessagesMetaPayload } from './fixtures';

describe('TopicMessages reducer', () => {
  it('returns the initial state', () => {
    expect(reducer(undefined, fetchSchemaVersionsAction.request())).toEqual(
      initialState
    );
  });
  it('Adds new message', () => {
    const state = reducer(undefined, addTopicMessage(topicMessagePayload));
    expect(state.messages.length).toEqual(1);
    expect(state).toMatchSnapshot();
  });
  it('Clears messages', () => {
    const state = reducer(undefined, addTopicMessage(topicMessagePayload));
    expect(state.messages.length).toEqual(1);

    const newState = reducer(state, resetTopicMessages());
    expect(newState.messages.length).toEqual(0);
  });
  it('Updates Topic Messages Phase', () => {
    const phase = 'Polling';

    const state = reducer(undefined, updateTopicMessagesPhase(phase));
    expect(state.phase).toEqual(phase);
  });
  it('Updates Topic Messages Meta', () => {
    const state = reducer(
      undefined,
      updateTopicMessagesMeta(topicMessagesMetaPayload)
    );
    expect(state.meta).toEqual(topicMessagesMetaPayload);
  });
});
