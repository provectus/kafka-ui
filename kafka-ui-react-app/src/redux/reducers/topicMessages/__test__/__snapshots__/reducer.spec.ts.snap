// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicMessages reducer Adds new message 1`] = `
Object {
  "messages": Array [
    Object {
      "content": "{\\"host\\":\\"schemaregistry1\\",\\"port\\":8085,\\"master_eligibility\\":true,\\"scheme\\":\\"http\\",\\"version\\":1}",
      "headers": Object {},
      "key": "schema-registry",
      "offset": 14,
      "partition": 29,
      "timestamp": 2021-07-21T23:25:14.865Z,
      "timestampType": "CREATE_TIME",
    },
  ],
  "meta": Object {
    "bytesConsumed": 0,
    "elapsedMs": 0,
    "isCancelled": false,
    "messagesConsumed": 0,
  },
}
`;
